File: src/main/java/net/cjsah/console/command/source/ConsoleCommandSource.java
package net.cjsah.console.command.source;

import net.cjsah.console.Console;
import net.cjsah.console.plugin.Plugin;
import org.apache.logging.log4j.Level;

public class ConsoleCommandSource extends CommandSource<Console> {

    public ConsoleCommandSource(Console source) {
        super(source);
    }

    @Override
    public boolean CanUse(Plugin plugin) {
        return true;
    }

    @Override
    public void sendFeedBack(String message) {
        this.sendFeedBack(message, Level.INFO);
    }

    @Override
    public void sendFeedBack(String message, Level level) {
        if (Level.INFO.equals(level)) {
            Console.INSTANCE.getLogger().info(message);
        } else if (Level.WARN.equals(level)) {
            Console.INSTANCE.getLogger().warn(message);
        } else if (Level.ERROR.equals(level)) {
            Console.INSTANCE.getLogger().error(message);
        } else if (Level.DEBUG.equals(level)) {
            Console.INSTANCE.getLogger().debug(message);
        }
    }
}

File: src/main/kotlin/net/cjsah/console/command/source/ConsoleCommandSource.kt
package net.cjsah.console.command.source

import net.cjsah.console.Console
import net.cjsah.console.Logger
import net.cjsah.console.Permissions
import net.cjsah.console.plugin.Plugin
import org.apache.logging.log4j.Level

class ConsoleCommandSource : CommandSource<Console>(Console) {
    override fun hasPermission(permission: Permissions.PermissionType) = true

    override fun canUse(plugin: Plugin) = true

    override suspend fun sendFeedBack(message: String) = sendFeedBack(message, Level.INFO)

    override suspend fun sendFeedBack(message: String, level: Level) {
        when (level) {
            Level.INFO -> Logger.info(message)
            Level.WARN -> Logger.warn(message)
            Level.ERROR -> Logger.error(message)
            Level.DEBUG -> Logger.debug(message)
        }
    }
}